src/executables/commdb.c

```diff
@@ -1075,8 +1091,20 @@ process_batch_command (CSS_CONN_ENTRY * conn)
 
   if ((commdb_Arg_server_name) && (!commdb_Arg_halt_shutdown))
     {
-      pid = process_server_info_pid (conn, (char *) commdb_Arg_server_name, COMM_SERVER);
-      process_slave_kill (conn, (char *) commdb_Arg_server_name, commdb_Arg_shutdown_time, pid);
+      if (!commdb_Arg_single_node)
+	{
+	  pid = process_server_info_pid (conn, (char *) commdb_Arg_server_name, COMM_SERVER, SERVER_TYPE_ANY);
+	  process_slave_kill (conn, (char *) commdb_Arg_server_name, commdb_Arg_shutdown_time, pid);
+	}
+      else
+	{
+	  // kill the transaction server first
+	  pid = process_server_info_pid (conn, (char *) commdb_Arg_server_name, COMM_SERVER, SERVER_TYPE_TRANSACTION);
```

Comment:
Same comment as on the start side. The system should be able to recover from any situation when only one of the servers is currently alive. It should also not try to stop a server which is running - on the same port - but is not in the 'single node' config (as in the situation where: server was started in single not config, config was changed, an now it is trying to stop servers in "single node" config).

Reply:
For the case that only one server is started when single_node configuration stop command is issued, to try and stop both servers in the right order is the best approach. A small change required here, pid must be verified; if it is equal to zero, calling process_slave_kill is not necessary.

The other case could be considered a problem:

- Both servers are started within single_node configuration
- Configuration is changed to something else.
- `cubrid server stop` command is issued. it will only try to stop one server.

The easiest approach may be to try and stop both type of servers regardless of configuration. In consequence, the only required change is how server stop is handled on commdb; the process_server stop changes and the --single-node-stop commdb option would become unnecessary.

Do you guys agree with this?

Reply:
As I mentioned in the starting servers discussion thread, handling the configuration changes is not really important. More important about the suggested change is that it simplifies this patch.

Reply:
I will change so that it always looks for both a transaction and a page server and only stops the ones with valid pids.

Reply:
@adrianbrab The suggestion was to always try to stop 3 processes: one regular server, two ATS/PS servers. Since they can only be started in the respective disjoint configurations, the stopping of 3 possible processes is a "catch all" option. I think it's ok. (please note that I do not know if stopping  an "any" server will also stop either a TS or PS whichever is found or will not stop anything; I am not familiar with that part).

Reply:
@cristiarg  wouldn't a search for both a TS and a PS always stop all the present servers? As an unknown server will default to be a TS.

Reply:
As said in my last remark of my previous comment, I am not familiar with that part.
If it suffices to search two times to stop any combination of:
- default cub_server started
- only TS started
- only PS started
- both TS and PS started
then that is what needs to be done.
Just make sure to have the scenario tested.

Reply:
Default cub_server is a transaction server. Stopping one transaction server and one page server should cover all mentioned cases.